### ISSUE TITLE
Can't detect non-compliant AC3 stream

### ISSUE THREAD

**Issue by lonecrane** [justdan96/tsMuxer#455]
Hi, I have encounted a strange bd source whose track makes tsMuxeR says "Can't detect stream type".
Track ID:    4354
The sample file has been uploaded to [sample](https://www.mediafire.com/file/f032nq6gfs2wukd/sample_BD.rar/file)
The lastest version makes no difference.
Thanks in advance for any help.

```
tsMuxeR version git-f4fe29f. github.com/justdan96/tsMuxer
HEVC muxing fps is not set. Get fps from stream. Value: 23.976
B-pyramid level 2 detected. Shift DTS to 3 frames
Track ID:    4113
Stream type: HEVC
Stream ID:   V_MPEGH/ISO/HEVC
Stream info: Profile: Main10@5.1 Resolution: 3840:2160p  Frame rate: 23.976
Stream lang: 

Track ID:    4352
Stream type: TRUE-HD
Stream ID:   A_AC3
Stream info: AC3 core+TRUE-HD. Peak bitrate: 5997Kbps (core 640Kbps) Sample Rate: 48KHz Channels: 7.1
Stream lang: eng

Track ID:    4353
Stream type: AC3
Stream ID:   A_AC3
Stream info: Bitrate: 320Kbps Sample Rate: 48KHz Channels: 2
Stream lang: eng

Track ID:    4354
Can't detect stream type

Track ID:    4355
Stream type: DTS-HD
Stream ID:   A_DTS
Stream info: Bitrate: 1536Kbps  core + MLP data.Sample Rate: 48KHz  Channels: 7.1 (DTS Master Audio)
Stream lang: zho
Stream delay: 4

Track ID:    4356
Stream type: AC3
Stream ID:   A_AC3
Stream info: Bitrate: 640Kbps Sample Rate: 48KHz Channels: 5.1
Stream lang: zho
Stream delay: 25

Track ID:    4768
Stream type: PGS
Stream ID:   S_HDMV/PGS
Stream info: Presentation Graphic Stream #0
Stream lang: eng

Track ID:    4769
Stream type: PGS
Stream ID:   S_HDMV/PGS
Stream info: Presentation Graphic Stream #1 Resolution: 1920:1080 Frame rate: 23.976
Stream lang: zho

Track ID:    4770
Stream type: PGS
Stream ID:   S_HDMV/PGS
Stream info: Presentation Graphic Stream #2 Resolution: 1920:1080 Frame rate: 23.976
Stream lang: zho

Track ID:    4771
Stream type: PGS
Stream ID:   S_HDMV/PGS
Stream info: Presentation Graphic Stream #3 Resolution: 1920:1080 Frame rate: 23.976
Stream lang: zho

Track ID:    4772
Stream type: PGS
Stream ID:   S_HDMV/PGS
Stream info: Presentation Graphic Stream #4 Resolution: 1920:1080 Frame rate: 23.976
Stream lang: zho

Track ID:    4773
Stream type: PGS
Stream ID:   S_HDMV/PGS
Stream info: Presentation Graphic Stream #5 Resolution: 1920:1080 Frame rate: 23.976
Stream lang: spa

Track ID:    4774
Stream type: PGS
Stream ID:   S_HDMV/PGS
Stream info: Presentation Graphic Stream #6 Resolution: 1920:1080 Frame rate: 23.976
Stream lang: jpn

File #00000 name=I:\Demux_to\BD_with_problem\BDMV\STREAM\00055.m2ts
Duration: 00:02:18.263
start-time: 188925000
Marks: 00:00:00.000 00:00:09.092 00:01:42.894 

File #00001 name=I:\Demux_to\BD_with_problem\BDMV\STREAM\00056.m2ts
Duration: 02:15:48.306
start-time: 188925000
Marks: 00:03:51.689 00:06:12.121 00:09:33.781 00:10:43.392 00:12:35.546 
Marks: 00:13:39.527 00:14:48.053 00:16:45.754 00:18:04.416 00:19:21.368 
Marks: 00:21:26.451 00:24:03.483 00:26:46.855 00:29:54.501 00:31:49.157 
Marks: 00:35:51.607 00:36:40.448 00:37:45.346 00:40:19.083 00:43:39.033 
Marks: 00:45:29.393 00:49:22.751 00:51:19.785 00:54:13.583 00:59:48.626 
Marks: 01:04:15.893 01:08:37.071 01:11:07.054 01:15:16.220 01:16:49.730 
Marks: 01:19:37.355 01:20:56.726 01:25:36.005 01:32:12.401 01:35:28.848 
Marks: 01:36:34.788 01:39:28.462 01:41:29.792 01:42:57.337 01:44:59.835 
Marks: 01:50:33.543 01:53:39.604 01:55:21.998 01:58:05.286 02:01:05.424 
Marks: 02:02:32.511 02:07:07.411 02:18:06.319 

```

**Comment by jcdr428** [comment]
@lonecrane this is not the original BD. The track 4352 (AC3 core+TrueHD) seems to have been split by the ripper into two tracks 4353 (AC3) and 4354 (MLP/TrueHD). A Blu-ray cannot have MLP/TrueHD tracks without an AC3 core.

**Comment by lonecrane** [comment]
It seems that the track is a thd+ac3 track rather than a pure thd one. Is it possible for you to have a look into it?

The info of this track reported by BDinfo:
`Dolby TrueHD Audio              Chinese         2603 kbps       7.1 / 48 kHz / 2155 kbps / 16-bit (AC3 Core: 5.1 / 48 kHz / 448 kbps)`

In addition, with another demux tool (dgdemux), this track is extracted as test.thd, and eac3to.exe also reports that it is a thd+ac3:
`"eac3to.exe" "H:\test.thd"
TrueHD/AC3, 7.1 channels, 48kHz
(embedded: AC3, 5.1 channels, 448kbps, 48kHz)`

**Comment by jcdr428** [comment]
@lonecrane I have analysed the stream 4354 in more detail: it is an ac3+thd stream, however the ac3 stream is non-compliant i.e. there is a syncword 0x0B77 which is not where it should be. This is why the stream is not detected as AC3 neither by MediaInfo not by tsMuxer.
As mentionned before, the bluray is not original, this can be seen in the playlist where all chinese streams don't appear nad have obviously been added.

**Comment by lonecrane** [comment]
@jcdr428 This disc is indeed a DIY one by a local group. There are some other discs with this stream type. As far as I know, these discs can be played normally by some popular players and boxes such as DVDFab player5.0.1.1、PD17.0.2101.62、Egreat A5/A10、OPPO203 and so on. Although I know that always tens of issues are calling for your help, could it be possible to make tsmuxer recognize this non-compliant stream type?

**Comment by jcdr428** [comment]
@lonecrane personally I won't look at it as I am not keen in allowing txMuxer to mux non-compliant streams; I will leave the issue open as somebody else might want to look at this issue.

**Comment by justdan96** [comment]
@jcdr428 Would you be able to suggest a new title for this issue?

**Comment by jcdr428** [comment]
@justdan96 What about "Can't detect non-compliant AC3 stream" ?

**Comment by jcdr428** [comment]
Fixed by be590646
